source/helpers.js

```diff
@@ -15,3 +36,123 @@ exports.generateTextBar = function (numerator, denominator, barLength) {
 	}
 	return bar;
 }
+
+exports.getNumberEmoji = function (number) {
+	switch (Number(number)) {
+		case 0:
+			return '0️⃣';
+		case 1:
+			return '1️⃣';
+		case 2:
+			return '2️⃣';
+		case 3:
+			return '3️⃣';
+		case 4:
+			return '4️⃣';
+		case 5:
+			return '5️⃣';
+		case 6:
+			return '6️⃣';
+		case 7:
+			return '7️⃣';
+		case 8:
+			return '8️⃣';
+		case 9:
+			return '9️⃣';
+		case 10:
+			return '🔟';
+		default:
+			return '#️⃣';
+	}
+}
+
+/** Recalculates the ranks (standard deviations from mean) and placements (ordinal) for the given participants
+ * @param {Hunter[]} participants
+ * @param {GuildRank[]} ranks
+ * @returns Promise of the message congratulating the hunter reaching first place (or `null` if no change)
+ */
+exports.setRanks = async (participants, ranks) => {
+	let previousFirstPlaceId;
+	let mean = 0;
+	const rankableHunters = [];
+	for (const hunter of participants) {
+		if (hunter.isRankEligible) {
+			if (hunter.seasonPlacement == 1) {
+				previousFirstPlaceId = hunter.userId;
+			}
+			hunter.lastRank = hunter.rank;
+			mean += hunter.seasonXP;
+			rankableHunters.push(hunter);
+		} else {
+			hunter.nextRankXP = 0;
+		}
+	}
+	const n = Math.max(rankableHunters.length, 2);
```

Comment:
...Does this fully make sense? What if there's only one rankable hunter? If this solves a weird edge case with exactly one hunter, I feel like that edge case would affect everything down the line to the point that we just want different handling for the only 1 hunter case, potentially skipping on ranks in the first place.

Reply:
Constraint: n = 1 causes a divide by 0 later, but we still want to touch everyone for the case of "first recalculation after a season reset".
Alleged previous thinking: I think we settled on this as a compromise because only one person being a season participant was something of a trivial case.

Haven't thought through what the functionality of a more formal special case would be though.